### ISSUE TITLE
Create Workbook from template

### ISSUE THREAD

**Issue by lscheffler** [ggreen86/XLSX-Workbook-Class#47]
How to create a new workbook from template? Load the template and save ander different name or si there a different way?

TIA

**Comment by ggreen86** [comment]
Yes, load the workbook and then make the changes and use the SaveWorkbookAs() method.

________________________________
From: Lutz Scheffler ***@***.***>
Sent: Thursday, May 13, 2021 12:55 PM
To: ggreen86/XLSX-Workbook-Class ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [ggreen86/XLSX-Workbook-Class] Create Workbook from template (#47)


How to create a new workbook from template? Load the template and save ander different name or si there a different way?

TIA

**Comment by jhernancanom** [comment]
Hi, Greg.

I am going to do it this way:

1. From VFP, I use this sentence " copy TEMPLATE.XLSx to
VENTAS_ENERO_2021.XLSx ".

2. since XLSX-Workbook-Class I load VENTAS_ENERO_2021.XLSx, and....

3. I make the changes I need.

4. I use SaveWorkbook() method.

Is this a right way?

I need to confirm because with r31 I was receiving wrong results (in the
way mentioned here) (hidden cols and hidden sheets appear visible, some
formulas get out & stay values, several formats -bold, color-- disappear),
and I am downloading last release to try.

Thanks.


*HERNAN CANO MARTINEZ*
Analista de Sistemas
Cel: 300-782.48.44.
WhatsApp: 300-782.48.44.

**Comment by ggreen86** [comment]
Hernan--

That way should work.  You can also open the template.xlsx file and then use the SaveWorkbookAs() method to save with the name "VENTAS_ENERO_2021.XLSx".

Please send me the template.xlsx file so that I can test with it to ensure it is being opened correctly and retaining the settings.

Thank you.

Greg

________________________________

**Comment by lscheffler** [comment]
@ggreen86 , my problem was rather to load an _XLST_ excel template instead to copy an _XLSX_. Scenario:
- customer provides a XLST template
   -  this comes with properties set (that's why #48 )
   -  there are bookmarks set (#44,  #45,  #46 )
- from that I determine which data is to be loaded into the workbook, and on what location.

So the customer can create a kind of header I fill in, and sets  for example the position of column names and where the data starts.

I know there are differences how EXCEL deals with XLST and XLSX, but I have no insight if this is internal to the file or something determined on how the file is opened.

**Comment by ggreen86** [comment]
Can you send a copy of the XLST template file?  I would like to look 'under the hood' to see what differences there are in it vs. the XLSX file.  Possibly this can be added to the class with the other needs below.

Greg

________________________________

**Comment by lscheffler** [comment]
Well, the simplest would be to create something on EXCEL and save as XLST? Possibly a file you have as XLSX, so you can compare?
Anyway, see zip attached. It's nothing then some cells dedicated to receive data bound to there named cells.

[Export.zip](https://github.com/ggreen86/XLSX-Workbook-Class/files/6480527/Export.zip)

**Comment by lscheffler** [comment]
Same as above, as xlsx and xltx
[Vorlagen.zip](https://github.com/ggreen86/XLSX-Workbook-Class/files/6480551/Vorlagen.zip)

**Comment by ggreen86** [comment]
The file you sent is a xlt (not a xltx).  It is the older format for the template file which is not supported (it is not an xml format that can be parsed).

________________________________

**Comment by lscheffler** [comment]
The [file](https://github.com/ggreen86/XLSX-Workbook-Class/files/6480551/Vorlagen.zip) in the [second message](https://github.com/ggreen86/XLSX-Workbook-Class/issues/47#issuecomment-841420644) comes with xltx. As far as I see it's xml in a zip.
